packages/asc-ui/src/components/Checkbox/Checkbox.tsx

```diff
@@ -40,13 +41,14 @@ const Checkbox = React.forwardRef<
     return (
       <CheckboxWrapperStyle
         {...{ className, disabled, focus, checked, error }}
+        // eslint-disable-next-line no-nested-ternary
         aria-disabled={disabled}
       >
         <CheckboxIconStyle {...{ disabled, variant, checked, error }} size={15}>
-          {checked && <Checkmark />}
+          {checked && (indeterminate ? <Indeterminate /> : <Checkmark />)}
         </CheckboxIconStyle>
         <CheckboxStyle
-          {...{ ...otherProps, disabled, checked, ref }}
+          {...{ ...otherProps, disabled, checked, indeterminate, ref }}
```

Comment:
`indeterminate` is a property on `HTMLInputElement`, not an attribute. This means that you cannot simply set it as a prop, you will have to use the `ref` of the element and set it there manually. For more information check out this blog [post](https://davidwalsh.name/react-indeterminate
).

Reply:
This one is not meant to be set as an attribute, but needed this for styling at first, but it's not needed anymore so I removed it :)